src/odemis/driver/smartpod.py

```diff
@@ -0,0 +1,835 @@
+        f = self._executor.submitf(f, self._doMoveRel, f, shift)
+        return f
+    
+    def _doMoveRel(self, future, shift):
+        """
+        Do a relative move by converting it into an absolute move
+        """
+        pos = add_coord(self.position.value, shift)
+        self._doMoveAbs(future, pos)
+
+# Only for testing/simulation purpose
+# Very rough version that is just enough so that if the wrapper behaves correctly,
+# it returns the expected values.
+
+
+def _deref(p, typep):
+    """
+    p (byref object)
+    typep (c_type): type of pointer
+    Use .value to change the value of the object
+    """
+    # This is using internal ctypes attributes, that might change in later
+    # versions. Ugly!
+    # Another possibility would be to redefine byref by identity function:
+    # byref= lambda x: x
+    # and then dereferencing would be also identity function.
+    return typep.from_address(addressof(p._obj))
+
+
+class FakeSmartPodDLL(object):
+    """
+    Fake smartpod DLL for simulator
+    """
+
+    def __init__(self):
+        self.pose = Pose()
+        self.target = Pose()
+        self.properties = {}
+        self._speed = c_double(0)
+        self._speed_control = c_int()
+        self.accel = c_double(0)
+        self.referenced = False
+
+        # Sepcify ranges
+        self._range = {}
+        self._range['x'] = (-1, 1)
+        self._range['y'] = (-1, 1)
+        self._range['z'] = (-1, 1)
+        self._range['theta_x'] = (-3.14, 3.14)
+        self._range['theta_y'] = (-3.14, 3.14)
+        self._range['theta_z'] = (-3.14, 3.14)
+
+        self.stopping = threading.Event()
+
+        self._current_move_start = time.time()
+        self._current_move_finish = time.time()
+
+    def _pose_in_range(self, pose):
+        if self._range['x'][0] <= pose.positionX <= self._range['x'][1] and \
+            self._range['y'][0] <= pose.positionY <= self._range['y'][1] and \
+            self._range['z'][0] <= pose.positionZ <= self._range['z'][1] and \
+            self._range['theta_x'][0] <= pose.rotationX <= self._range['theta_x'][1] and \
+            self._range['theta_y'][0] <= pose.rotationY <= self._range['theta_y'][1] and \
+            self._range['theta_z'][0] <= pose.rotationZ <= self._range['theta_z'][1]:
+            return True
+        else:
+            return False
+
+    """
+    DLL functions (fake)
+    These functions are provided by the real SmartPod DLL
+    """
+    def Smarpod_Open(self, id, timeout, locator, options):
+        return SmartPodDLL.SMARPOD_OK.value
+
+    def Smarpod_Close(self, id):
+        return SmartPodDLL.SMARPOD_OK.value
+
+    def Smarpod_SetSensorMode(self, id, mode):
+        return SmartPodDLL.SMARPOD_OK.value
+
+    def Smarpod_FindReferenceMarks(self, id):
+        self.stopping.clear()
+        time.sleep(0.5)
+        if self.stopping.is_set():
+            self.referenced = False
+            return SmartPodDLL.SMARPOD_STOPPED_ERROR.value
```

Comment:
These functions simulate the DLL, including the errcheck wrapping. So instead of returning an error, it should raise a SmartPodError.